src/framework/auth/services/token/token-packer.ts

```diff
@@ -0,0 +1,44 @@
+import { Inject, Injectable } from '@angular/core';
+
+import { NB_AUTH_TOKENS } from '../../auth.options';
+import { NbAuthSimpleToken, NbAuthToken, nbCreateToken } from './token';
+
+interface NbTokenPack {
+  class: string,
+  value: string,
+}
+
+/**
+ *
+ */
+@Injectable()
+export class NbAuthTokenPacker {
+
+  constructor(@Inject(NB_AUTH_TOKENS) private availableTokens) {
+  }
+
+  pack(token: NbAuthToken): string {
+    return JSON.stringify({
+      class: token.constructor.name,
+      value: token.toString(),
+    });
+  }
+
+  unpack(rawToken: string): NbAuthToken {
+    let tokenPack: NbTokenPack;
+    try {
+      tokenPack = JSON.parse(rawToken);
+    } catch (e) {
+      console.warn('NbAuthToken: Can not unpack token, using empty `NbAuthSimpleToken`.');
```

Comment:
Maybe we have to throw an error and stop *unpack* execution? I don't think it's correct to warn and return an empty token.

Reply:
Let's add a token status (empty, invalid, valid, expired) or in the next version think about throwing an error.